### PR TITLE
Docs improvements and prep for deduping with docs repo

### PR DESCRIPTION
**Pull request by yorhodes**
Fixes https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/1812

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,15 +1,121 @@
...
+    end
+
+
+    subgraph "Gnosis"
+        HYP_G[TokenRouter]
+        style HYP_G fill:purple
+        Mailbox_G[(Mailbox)]
+    end
+
+    HYP_E -. "router" .- HYP_P -. "router" .- HYP_G
```
Comment (nambrot) [inline README.md:40]: Like I mentioned offline, i would replace `router` with `Hyperlane message` or something

File: README.md [hunk 2]
```diff
@@ -1,15 +1,121 @@
...
+    "mainBkg": "#025AA1",
+    "textColor": "white",
+    "clusterBkg": "beige"
+  },
+  "themeCSS": ".edgeLabel { color: black }"
+}}%%
+
+graph LR
+    subgraph "Ethereum"
+        HYP_E[TokenRouter]
```
Comment (nambrot) [inline README.md:22]: IDK if you were planning on doing this in a follow on change, but should this be tokenRouter, or the specific instances (HypNative/HypErc20Collateral)

File: README.md [hunk 3]
```diff
@@ -1,15 +1,121 @@
...
+
+To initiate a remote transfer, users call the `TokenRouter.transferRemote` function with the destination chain ID, recipient address, and transfer amount. 
+
+<details>
+<summary>Native Remote Transfer</summary>
+
+### Transfer Alice's `amount` native ETH from Ethereum to Bob on Polygon
+
+```mermaid
+%%{ init: {
```
Comment (nambrot) [inline README.md:60]: Does not render

File: README.md [hunk 4]
```diff
@@ -1,15 +1,121 @@
...
+    Bob((Bob))
+    style Bob fill:black
+    Alice((Alice))
+    style Alice fill:black
+
+    Relayer([Relayer])
+
+    subgraph "Ethereum"
+        HYP_E[NativeTokenRouter]
+        IGP_E[InterchainGasPaymaster]
```
Comment (nambrot) [inline README.md:80]: I would personally omit IGP, Relayers here. Maybe even ISMs and just point folks to SC/ISM documentation

File: contracts/libs/TokenRouter.sol [hunk 5]
```diff
@@ -61,6 +62,14 @@ abstract contract TokenRouter is GasRouter {
         emit SentTransferRemote(_destination, _recipient, _amountOrId);
     }
 
+    function transferRemote(
```
Comment (nambrot) [inline contracts/libs/TokenRouter.sol:65]: Nice sneaky change

**Review by nambrot — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -1,51 +1,236 @@
...
+        HYP_G[TokenRouter]
+        style HYP_G fill:orange
+        Mailbox_G[(Mailbox)]
+    end
+
+    HYP_E -. "router" .- HYP_P -. "router" .- HYP_G
+
+```
+
+The Token Router contract comes in several flavors and a warp route can be composed as  a combination of these flavors.
```
Comment (jmrossy) [inline README.md:44]: double space after as

File: contracts/libs/TokenRouter.sol (hunk 5 above)
Comment (jmrossy) [reply contracts/libs/TokenRouter.sol:65]: It is nice but it may cause confusion that some Hyp contracts take bytes32 and others take address

**Review by jmrossy — APPROVED** [review]

File: contracts/libs/TokenRouter.sol (hunk 5 above)
Comment (yorhodes) [reply contracts/libs/TokenRouter.sol:65]: reverting

**Pull request merged**